### PR TITLE
Adding custom initialization method to model constructor

### PR DESCRIPTION
**Pull request by dsgouda**
This is to mitigate issues found in https://github.com/azure/autorest/issues/781
The partial method CustomInit() will allow users to define their own defaults if necessary while at the same time does introduce any breaking changes
@olydis @fearthecowboy PTAL

### CODE REVIEW

**Comment by dsgouda** [comment]
@azuresdkci Test this please

**Comment by dsgouda** [comment]
@azuresdkci Test this please

**Comment by msftgits** [comment]
Closing and re-opening to ping CLA bot.

File: src/generator/AutoRest.CSharp/Templates/Rest/Common/ModelTemplate.cshtml [hunk 1]
```diff
@@ -118,9 +122,15 @@ namespace @(Settings.Namespace).@(Settings.ModelsName)
             {
             @:@(property.Name) = @(property.DefaultValue);
             }
+
         @:}
         }
-
+        @EmptyLine
+        /// <summary>
+        /// an Init method that performs custom operations like setting defaults
```
Comment (olydis) [inline src/generator/AutoRest.CSharp/Templates/Rest/Common/ModelTemplate.cshtml]: please improve summary text
- uppercase first letter
- `Init` => `initialization` or something (`Init` sounds like another method name)

**Comment by dsgouda** [comment]
@azuresdkci Test this please

**Pull request merged**